src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1884,5 +1894,27 @@ internal void EnsureDocumentOptionProvidersInitialized()
                 }
             }
         }
+
+#nullable enable
+
+        internal void RefreshProjectExistsUIContextForLanguage(string language)
+        {
+            lock (_gate)
+            {
+                var uiContext =
+                    _languageToProjectExistsUIContext.GetOrAdd(
+                        language,
+                        l => Services.GetLanguageServices(l).GetService<IProjectExistsUIContextProviderLanguageService>()?.GetUIContext());
+
+                if (uiContext != null)
+                {
+                    var projectsWithLanguageName = CurrentSolution.Projects.Count(p => p.Language == language);
+
+                    uiContext.IsActive = (projectsWithLanguageName > 0);
```

Comment:
@ryanmolden does this need to be on the UI thread to set the active state?

Reply:
I am hoping the event handlers for consumers of this UI Context will still be fired on the UI thread (we rely on that to avoid races where event fires while the event handler is being hooked up). I'll build this PR locally and test this with UT bits to confirm...

Reply:
Verified that event handler still fires on UI thread.

Reply:
@ryanmolden confirmed privately that setting IsActive will do a COM RPC which we obviously don't want to do. Code is now updated to use JTF to asynchronously defer the work to the UI thread if we aren't shutting down the project on the UI thread. (If we are, it'll run synchronously and that's fine.)